### PR TITLE
168 benchmarks add the statlog german credit dataset

### PR DESCRIPTION
**Pull request by kmariuszk**
This pull request adds the Statlog German Credit dataset to the package, incorporating the necessary changes and updates across multiple files.

To accomplish this, the following tasks have been completed:

- Implemented loader functions in the `src\data\tabular.jl` file to handle the loading of the Statlog German Credit dataset.
- Exported the implemented functions in the `src\data\Data.jl` file to make them accessible to other parts of the package.
- Conducted thorough testing to ensure the correctness and reliability of the implemented functions.
- Updated the dataset notebook at `dev\artifacts\tabular.qmd` to reflect the inclusion of the Statlog German Credit dataset.
- Mentioned the addition of the Statlog German Credit dataset in the `students_documentation\summary.qmd` file to provide relevant information to users.
- Updated the `students_documentation\README.md` file to include a reference to the newly added dataset.
- Added examples that demonstrate the usage of the Statlog German Credit dataset in the `students_documentation\summary.qmd` file to aid users in understanding and utilizing the dataset effectively.
- Included documentation about the Statlog German Credit dataset in the `docs/src/tutorials/data_catalogue/` section to provide comprehensive information and usage guidelines.
- Added the Statlog German Credit dataset to the release artifacts on the https://github.com/JuliaTrustworthyAI/CounterfactualExplanations.jl/releases/tag/artifacts-1.8 page.

### CODE REVIEW

File: src/data/tabular.jl [hunk 1]
```diff
@@ -111,3 +111,54 @@ function load_credit_default(n::Union{Nothing,Int}=5000)
...
+# Arguments
+- `n::Union{Nothing, Int}=nothing`: The number of samples to subsample from the dataset. If `n` is not specified, all samples will be used. Must be <= 1000 and >= 1.
+
+# Returns
+- `counterfactual_data::CounterfactualData`: A `CounterfactualData` object containing the preprocessed data.
+
+# Example
+data = load_german_credit(500) # loads and preprocesses 500 samples from the German Credit dataset
+
+"""
```
Comment (RaunoArike) [inline src/data/tabular.jl:129]: I like this way of clearly specifying the arguments and the return value and giving an example - we should use this kind of documentation in other PRs as well.

File: src/data/tabular.jl [hunk 2]
```diff
@@ -111,3 +111,54 @@ function load_credit_default(n::Union{Nothing,Int}=5000)
...
+function load_german_credit(n::Union{Nothing,Int}=nothing)
+    # Throw an exceptoin if n > 1000:
+    if !isnothing(n) && n > 1000
+        throw(ArgumentError("n must be <= 1000"))
+    end
+
+    # Throw an exceptoin if n < 1:
```
Comment (RaunoArike) [inline src/data/tabular.jl:136]: Small typo - exceptoin

File: src/data/tabular.jl (hunk 1 above)
Comment (RaunoArike) [inline src/data/tabular.jl:129]: I like this way of clearly specifying the arguments and the return value and giving an example - we should use this kind of documentation in other PRs as well.

Comment (kmariuszk) [reply src/data/tabular.jl:129]: I'm glad to hear that 👍

File: test/data.jl [hunk 3]
```diff
@@ -40,3 +40,23 @@ end
...
+    counterfactual_data = load_german_credit(500)
+    @test size(counterfactual_data.X)[2] == 500
+    @test size(counterfactual_data.X)[1] == 20
+    @test size(counterfactual_data.y)[2] == 500
+
+    # Test case: Load data with n > 1000, expecting an error
+    @test_throws ArgumentError load_german_credit(1500)
+
+    # Test case: Load data with n < 1, expecting an error
+    @test_throws ArgumentError load_german_credit(0)
```
Comment (RaunoArike) [inline test/data.jl:61]: We could also add a test with a negative n to make the code even more robust to future changes.

**Review by RaunoArike — APPROVED** [review]
Good job! Left a couple of comments, but those are very small things, so I'll approve.

File: src/data/tabular.jl (hunk 2 above)
Comment (kmariuszk) [reply src/data/tabular.jl]: Fixed.

**Review by RaunoArike — APPROVED** [review]

**Review by laurikskl — APPROVED** [review]
Looks good to me, no major changes since my last review and I agree that this does make it tidier, nice to see the artifacts added as well! Rauno mentioned the only thing I wanted to nit-pick about already.
